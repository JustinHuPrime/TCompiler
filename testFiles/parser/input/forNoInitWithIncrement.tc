module foo;

void bar(int i) {
  for (; i > 0; --i) --i;
}