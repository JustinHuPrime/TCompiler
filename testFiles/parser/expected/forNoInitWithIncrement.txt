testFiles/parser/forNoInitWithIncrement.tc (code):
FILE(1, 1, STAB(ENTRY(bar, FUNCTION(testFiles/parser/forNoInitWithIncrement.tc, 3, 1, void(int)))), MODULE(1, 1, ID(1, 8, foo, REFERENCES())), FUNDEFN(3, 1, KEYWORDTYPE(3, 1, void), ID(3, 6, bar, REFERENCES(testFiles/parser/forNoInitWithIncrement.tc, 3, 1)), KEYWORDTYPE(3, 10, int), ID(3, 14, i, REFERENCES()), STAB(ENTRY(i, VARIABLE(testFiles/parser/forNoInitWithIncrement.tc, 3, 10, int))), COMPOUNDSTMT(3, 17, STAB(), FORSTMT(4, 3, STAB(), NULLSTMT(4, 8), BINOPEXP(4, 10, GT, ID(4, 10, i, REFERENCES(testFiles/parser/forNoInitWithIncrement.tc, 3, 10)), LITERAL(4, 14, UBYTE(0))), UNOPEXP(4, 17, PREDEC, ID(4, 19, i, REFERENCES(testFiles/parser/forNoInitWithIncrement.tc, 3, 10))), EXPRESSIONSTMT(4, 22, UNOPEXP(4, 22, PREDEC, ID(4, 24, i, REFERENCES(testFiles/parser/forNoInitWithIncrement.tc, 3, 10)))), STAB()))))
